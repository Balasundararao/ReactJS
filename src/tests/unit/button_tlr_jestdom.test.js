import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../../../src/components/practicecomponents/react_regular/button/button';

// testing-library....
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

//snapshot testing....
import renderer from "react-test-renderer";

afterEach(cleanup);
it('Renders with out crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render('<Button>', div);
})

it('Renders correctly', () => {
    const {getByTestId} = render(<Button label="click me please"></Button>)
    expect( getByTestId('button')).toHaveTextContent("click me please");
})

it('Renders correctly', () => {
    const {getByTestId} = render(<Button label="Save"></Button>)
    expect( getByTestId('button')).toHaveTextContent("Save");
})

it('Matches SnapShot ', () => {
    const tree = renderer.create(<Button label="Save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})


