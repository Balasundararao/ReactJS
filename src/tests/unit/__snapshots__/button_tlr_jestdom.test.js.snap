// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches SnapShot  1`] = `
<div
  data-testid="button"
>
  Hi 
  Save
</div>
`;
